backend/inspirehep/search/api.py

```diff
@@ -185,16 +187,25 @@ def execute(self, *args, **kwargs):
                 LOGGER.exception("Match reference error.", query_string=query_string)
         return results
 
-    def query_for_superuser_or_users(self, query_string):
+    def query_by_user_role(self, query_string):
```

Comment:
This function became very complex with lots of indentation levels, I think that it might be good to split it in parts